occur "undefined reference to `faiss::gpu::randVal(int, int)'" when make test_gpu

# Summary

When I make test_gpu under /faiss, i meet 'undefined reference to `faiss::gpu' error
below is the trace log

make -C gpu/test run
make[1]: Entering directory '/home/ubuntu/faiss/gpu/test'
g++ -std=c++11 -I/usr/local/cuda-9.0/include -fPIC -m64 -Wall -g -O3 -fopenmp -Wno-sign-compare -DFINTEGER=int -fPIC -fopenmp -L/usr/local/cuda-9.0/lib64  TestGpuIndexFlat.cpp   -o TestGpuIndexFlat
TestGpuIndexFlat.cpp: In member function ‘virtual void TestGpuIndexFlat_CopyFrom_Test::TestBody()’:
TestGpuIndexFlat.cpp:256:7: warning: unused variable ‘device’ [-Wunused-variable]
   int device = faiss::gpu::randVal(0, faiss::gpu::getNumDevices() - 1);
       ^
/tmp/cckWJeWo.o: In function `testFlat(TestFlatOptions const&)':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:55: undefined reference to `faiss::gpu::randVal(int, int)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlatIP::IndexFlatIP(long)':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:73: undefined reference to `faiss::IndexFlat::IndexFlat(long, faiss::MetricType)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlatL2::IndexFlatL2(long)':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:79: undefined reference to `faiss::IndexFlat::IndexFlat(long, faiss::MetricType)'
/tmp/cckWJeWo.o: In function `testFlat(TestFlatOptions const&)':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:70: undefined reference to `faiss::gpu::getNumDevices()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:70: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:72: undefined reference to `faiss::gpu::StandardGpuResources::StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:73: undefined reference to `faiss::gpu::StandardGpuResources::noTempMemory()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:81: undefined reference to `faiss::gpu::GpuIndexFlatIP::GpuIndexFlatIP(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:82: undefined reference to `faiss::gpu::GpuIndexFlatL2::GpuIndexFlatL2(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:88: undefined reference to `faiss::gpu::randVecs(unsigned long, unsigned long)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:56: undefined reference to `faiss::gpu::getMaxKSelection()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:56: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:108: undefined reference to `faiss::gpu::compareIndices(faiss::Index&, faiss::Index&, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, float, float, float)'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatIP::~GpuIndexFlatIP()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:159: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `testFlat(TestFlatOptions const&)':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:72: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `testFlat(TestFlatOptions const&)':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:45: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:47: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:49: undefined reference to `faiss::gpu::randVal(int, int)'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatIP::~GpuIndexFlatIP()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:159: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `testFlat(TestFlatOptions const&)':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:72: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_QueryEmpty_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:212: undefined reference to `faiss::gpu::StandardGpuResources::StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:213: undefined reference to `faiss::gpu::StandardGpuResources::noTempMemory()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:221: undefined reference to `faiss::gpu::GpuIndexFlatL2::GpuIndexFlatL2(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:232: undefined reference to `faiss::gpu::GpuIndex::search(long, float const*, long, float*, long*) const'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<float, float>(char const*, char const*, float const&, float const&)':
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_QueryEmpty_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:235: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:235: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:235: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:235: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<long, int>(char const*, char const*, long const&, int const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_QueryEmpty_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:239: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:239: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:239: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:239: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_QueryEmpty_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:212: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<float, float>(char const*, char const*, float const&, float const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<long, int>(char const*, char const*, long const&, int const&)':
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_QueryEmpty_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:212: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:239: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:235: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:244: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:245: undefined reference to `faiss::gpu::randVal(int, int)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlatL2::IndexFlatL2(long)':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:79: undefined reference to `faiss::IndexFlat::IndexFlat(long, faiss::MetricType)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:249: undefined reference to `faiss::gpu::randVecs(unsigned long, unsigned long)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:250: undefined reference to `faiss::IndexFlat::add(long, float const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:252: undefined reference to `faiss::gpu::StandardGpuResources::StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:253: undefined reference to `faiss::gpu::StandardGpuResources::noTempMemory()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:256: undefined reference to `faiss::gpu::getNumDevices()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:256: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:263: undefined reference to `faiss::gpu::GpuIndexFlatL2::GpuIndexFlatL2(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:264: undefined reference to `faiss::gpu::GpuIndexFlatL2::copyFrom(faiss::IndexFlatL2*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:266: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:266: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:266: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:266: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:267: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:267: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:267: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:267: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:269: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:269: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:269: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:269: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:270: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:270: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:270: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:270: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:272: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:275: undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct(long, float*) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:278: undefined reference to `faiss::IndexFlat::reconstruct(long, float*) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:280: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:280: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:280: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:280: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:252: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:252: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:267: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:280: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:266: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:269: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:270: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:284: undefined reference to `faiss::gpu::StandardGpuResources::StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:285: undefined reference to `faiss::gpu::StandardGpuResources::noTempMemory()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:287: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:288: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:290: undefined reference to `faiss::gpu::getNumDevices()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:290: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:297: undefined reference to `faiss::gpu::GpuIndexFlatL2::GpuIndexFlatL2(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:299: undefined reference to `faiss::gpu::randVecs(unsigned long, unsigned long)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:300: undefined reference to `faiss::gpu::GpuIndexFlat::add(long, float const*)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlatL2::IndexFlatL2(long)':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:79: undefined reference to `faiss::IndexFlat::IndexFlat(long, faiss::MetricType)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:304: undefined reference to `faiss::gpu::GpuIndexFlatL2::copyTo(faiss::IndexFlatL2*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:306: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:306: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:306: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:306: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:307: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:307: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:307: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:307: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:309: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:309: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:309: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:309: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:310: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:310: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:310: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:310: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:312: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:315: undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct(long, float*) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:318: undefined reference to `faiss::IndexFlat::reconstruct(long, float*) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:320: undefined reference to `testing::Message::Message()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:320: undefined reference to `testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:320: undefined reference to `testing::internal::AssertHelper::operator=(testing::Message const&) const'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:320: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:284: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:284: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:307: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:320: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:306: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:309: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:310: undefined reference to `testing::internal::AssertHelper::~AssertHelper()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:326: undefined reference to `faiss::gpu::getNumDevices()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:326: undefined reference to `faiss::gpu::randVal(int, int)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:328: undefined reference to `faiss::gpu::getFullUnifiedMemSupport(int)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlatL2::IndexFlatL2(long)':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:79: undefined reference to `faiss::IndexFlat::IndexFlat(long, faiss::MetricType)'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:343: undefined reference to `faiss::gpu::StandardGpuResources::StandardGpuResources()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:344: undefined reference to `faiss::gpu::StandardGpuResources::noTempMemory()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:350: undefined reference to `faiss::gpu::GpuIndexFlatL2::GpuIndexFlatL2(faiss::gpu::GpuResources*, int, faiss::gpu::GpuIndexFlatConfig)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:352: undefined reference to `faiss::gpu::randVecs(unsigned long, unsigned long)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:353: undefined reference to `faiss::IndexFlat::add(long, float const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:354: undefined reference to `faiss::gpu::GpuIndexFlat::add(long, float const*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:362: undefined reference to `faiss::gpu::compareIndices(faiss::Index&, faiss::Index&, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, float, float, float)'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:343: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::TestBody()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:343: undefined reference to `faiss::gpu::StandardGpuResources::~StandardGpuResources()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatL2::~GpuIndexFlatL2()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:134: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatIP::~GpuIndexFlatIP()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:159: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `faiss::gpu::GpuIndexFlatIP::~GpuIndexFlatIP()':
/home/ubuntu/faiss/gpu/test/../GpuIndexFlat.h:159: undefined reference to `faiss::gpu::GpuIndexFlat::~GpuIndexFlat()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::~TestGpuIndexFlat_UnifiedMemory_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:323: undefined reference to `testing::Test::~Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_UnifiedMemory_Test::~TestGpuIndexFlat_UnifiedMemory_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:323: undefined reference to `testing::Test::~Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::~TestGpuIndexFlat_CopyTo_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:283: undefined reference to `testing::Test::~Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyTo_Test::~TestGpuIndexFlat_CopyTo_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:283: undefined reference to `testing::Test::~Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_CopyFrom_Test::~TestGpuIndexFlat_CopyFrom_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:243: undefined reference to `testing::Test::~Test()'
/tmp/cckWJeWo.o:/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:243: more undefined references to `testing::Test::~Test()' follow
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_IP_Float32_Test::TestGpuIndexFlat_IP_Float32_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:111: undefined reference to `testing::Test::Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_L2_Float32_Test::TestGpuIndexFlat_L2_Float32_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:125: undefined reference to `testing::Test::Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_L2_Float32_K1_Test::TestGpuIndexFlat_L2_Float32_K1_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:140: undefined reference to `testing::Test::Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_IP_Float16_Test::TestGpuIndexFlat_IP_Float16_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:152: undefined reference to `testing::Test::Test()'
/tmp/cckWJeWo.o: In function `TestGpuIndexFlat_L2_Float16_Test::TestGpuIndexFlat_L2_Float16_Test()':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:166: undefined reference to `testing::Test::Test()'
/tmp/cckWJeWo.o:/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:181: more undefined references to `testing::Test::Test()' follow
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `testing::internal::scoped_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::reset(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/usr/include/gtest/internal/gtest-port.h:912: undefined reference to `testing::internal::IsTrue(bool)'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `faiss::IndexFlat::~IndexFlat()':
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `vtable for faiss::IndexFlat'
/home/ubuntu/faiss/gpu/test/../../IndexFlat.h:22: undefined reference to `faiss::Index::~Index()'
/tmp/cckWJeWo.o: In function `main':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:366: undefined reference to `testing::InitGoogleTest(int*, char**)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:369: undefined reference to `faiss::gpu::setTestSeed(long)'
/tmp/cckWJeWo.o: In function `RUN_ALL_TESTS()':
/usr/include/gtest/gtest.h:2288: undefined reference to `testing::UnitTest::GetInstance()'
/usr/include/gtest/gtest.h:2288: undefined reference to `testing::UnitTest::Run()'
/tmp/cckWJeWo.o: In function `__static_initialization_and_destruction_0':
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:111: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:111: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:125: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:125: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:140: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:140: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:152: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:152: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:166: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:166: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:181: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:181: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:194: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:194: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:211: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:211: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:243: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:243: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:283: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:283: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:323: undefined reference to `testing::internal::GetTestTypeId()'
/home/ubuntu/faiss/gpu/test/TestGpuIndexFlat.cpp:323: undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<long, int>(char const*, char const*, long const&, int const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<long, long>(char const*, char const*, long const&, long const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<int, int>(char const*, char const*, int const&, int const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o: In function `testing::AssertionResult testing::internal::CmpHelperEQ<std::vector<float, std::allocator<float> >, std::vector<float, std::allocator<float> > >(char const*, char const*, std::vector<float, std::allocator<float> > const&, std::vector<float, std::allocator<float> > const&)':
/usr/include/gtest/gtest.h:1460: undefined reference to `testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/include/gtest/gtest.h:1449: undefined reference to `testing::AssertionSuccess()'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTIN5faiss11IndexFlatIPE[_ZTIN5faiss11IndexFlatIPE]+0x10): undefined reference to `typeinfo for faiss::IndexFlat'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTIN5faiss11IndexFlatL2E[_ZTIN5faiss11IndexFlatL2E]+0x10): undefined reference to `typeinfo for faiss::IndexFlat'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTIN5faiss3gpu14GpuIndexFlatL2E[_ZTIN5faiss3gpu14GpuIndexFlatL2E]+0x10): undefined reference to `typeinfo for faiss::gpu::GpuIndexFlat'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTIN5faiss3gpu14GpuIndexFlatIPE[_ZTIN5faiss3gpu14GpuIndexFlatIPE]+0x10): undefined reference to `typeinfo for faiss::gpu::GpuIndexFlat'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI32TestGpuIndexFlat_IP_Float32_Test[_ZTI32TestGpuIndexFlat_IP_Float32_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI32TestGpuIndexFlat_L2_Float32_Test[_ZTI32TestGpuIndexFlat_L2_Float32_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI35TestGpuIndexFlat_L2_Float32_K1_Test[_ZTI35TestGpuIndexFlat_L2_Float32_K1_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI32TestGpuIndexFlat_IP_Float16_Test[_ZTI32TestGpuIndexFlat_IP_Float16_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI32TestGpuIndexFlat_L2_Float16_Test[_ZTI32TestGpuIndexFlat_L2_Float16_Test]+0x10): undefined reference to `typeinfo for testing::Test'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTI35TestGpuIndexFlat_L2_Float16_K1_Test[_ZTI35TestGpuIndexFlat_L2_Float16_K1_Test]+0x10): more undefined references to `typeinfo for testing::Test' follow
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x20): undefined reference to `faiss::Index::train(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x28): undefined reference to `faiss::IndexFlat::add(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x30): undefined reference to `faiss::Index::add_with_ids(long, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x38): undefined reference to `faiss::IndexFlat::search(long, float const*, long, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x40): undefined reference to `faiss::IndexFlat::range_search(long, float const*, float, faiss::RangeSearchResult*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x48): undefined reference to `faiss::IndexFlat::reset()'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x50): undefined reference to `faiss::IndexFlat::remove_ids(faiss::IDSelector const&)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x58): undefined reference to `faiss::IndexFlat::reconstruct(long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x60): undefined reference to `faiss::Index::reconstruct_n(long, long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatIPE[_ZTVN5faiss11IndexFlatIPE]+0x68): undefined reference to `faiss::Index::search_and_reconstruct(long, float const*, long, float*, long*, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x20): undefined reference to `faiss::Index::train(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x28): undefined reference to `faiss::IndexFlat::add(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x30): undefined reference to `faiss::Index::add_with_ids(long, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x38): undefined reference to `faiss::IndexFlat::search(long, float const*, long, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x40): undefined reference to `faiss::IndexFlat::range_search(long, float const*, float, faiss::RangeSearchResult*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x48): undefined reference to `faiss::IndexFlat::reset()'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x50): undefined reference to `faiss::IndexFlat::remove_ids(faiss::IDSelector const&)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x58): undefined reference to `faiss::IndexFlat::reconstruct(long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x60): undefined reference to `faiss::Index::reconstruct_n(long, long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss11IndexFlatL2E[_ZTVN5faiss11IndexFlatL2E]+0x68): undefined reference to `faiss::Index::search_and_reconstruct(long, float const*, long, float*, long*, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x20): undefined reference to `faiss::gpu::GpuIndexFlat::train(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x28): undefined reference to `faiss::gpu::GpuIndexFlat::add(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x30): undefined reference to `faiss::gpu::GpuIndex::add_with_ids(long, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x38): undefined reference to `faiss::gpu::GpuIndex::search(long, float const*, long, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x40): undefined reference to `faiss::Index::range_search(long, float const*, float, faiss::RangeSearchResult*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x48): undefined reference to `faiss::gpu::GpuIndexFlat::reset()'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x50): undefined reference to `faiss::Index::remove_ids(faiss::IDSelector const&)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x58): undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct(long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x60): undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct_n(long, long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x68): undefined reference to `faiss::Index::search_and_reconstruct(long, float const*, long, float*, long*, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x70): undefined reference to `faiss::gpu::GpuIndexFlat::addImplRequiresIDs_() const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x78): undefined reference to `faiss::gpu::GpuIndexFlat::addImpl_(int, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatL2E[_ZTVN5faiss3gpu14GpuIndexFlatL2E]+0x80): undefined reference to `faiss::gpu::GpuIndexFlat::searchImpl_(int, float const*, int, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x20): undefined reference to `faiss::gpu::GpuIndexFlat::train(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x28): undefined reference to `faiss::gpu::GpuIndexFlat::add(long, float const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x30): undefined reference to `faiss::gpu::GpuIndex::add_with_ids(long, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x38): undefined reference to `faiss::gpu::GpuIndex::search(long, float const*, long, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x40): undefined reference to `faiss::Index::range_search(long, float const*, float, faiss::RangeSearchResult*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x48): undefined reference to `faiss::gpu::GpuIndexFlat::reset()'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x50): undefined reference to `faiss::Index::remove_ids(faiss::IDSelector const&)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x58): undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct(long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x60): undefined reference to `faiss::gpu::GpuIndexFlat::reconstruct_n(long, long, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x68): undefined reference to `faiss::Index::search_and_reconstruct(long, float const*, long, float*, long*, float*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x70): undefined reference to `faiss::gpu::GpuIndexFlat::addImplRequiresIDs_() const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x78): undefined reference to `faiss::gpu::GpuIndexFlat::addImpl_(int, float const*, long const*)'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTVN5faiss3gpu14GpuIndexFlatIPE[_ZTVN5faiss3gpu14GpuIndexFlatIPE]+0x80): undefined reference to `faiss::gpu::GpuIndexFlat::searchImpl_(int, float const*, int, float*, long*) const'
/tmp/cckWJeWo.o:(.data.rel.ro._ZTV32TestGpuIndexFlat_IP_Float32_Test[_ZTV32TestGpuIndexFlat_IP_Float32_Test]+0x20): undefined reference to `t
....
....
collect2: error: ld returned 1 exit status
/tmp/cc0uePii.o:(.data.rel.ro._ZTV30TestGpuIndexFlat_CopyFrom_Test[_ZTV30TestGpuIndexFlat_CopyFrom_Test]+0x20): undefined reference to `testing::Test::SetUp()'
/tmp/cc0uePii.o:(.data.rel.ro._ZTV30TestGpuIndexFlat_CopyFrom_Test[_ZTV30TestGpuIndexFlat_CopyFrom_Test]+0x28): undefined reference to `testing::Test::TearDown()'
/tmp/cc0uePii.o:(.data.rel.ro._ZTV28TestGpuIndexFlat_CopyTo_Test[_ZTV28TestGpuIndexFlat_CopyTo_Test]+0x20): undefined reference to `testing::Test::SetUp()'
/tmp/cc0uePii.o:(.data.rel.ro._ZTV28TestGpuIndexFlat_CopyTo_Test[_ZTV28TestGpuIndexFlat_CopyTo_Test]+0x28): undefined reference to `testing::Test::TearDown()'
/tmp/cc0uePii.o:(.data.rel.ro._ZTV35TestGpuIndexFlat_UnifiedMemory_Test[_ZTV35TestGpuIndexFlat_UnifiedMemory_Test]+0x20): undefined reference to `testing::Test::SetUp()'
/tmp/cc0uePii.o:(.data.rel.ro._ZTV35TestGpuIndexFlat_UnifiedMemory_Test[_ZTV35TestGpuIndexFlat_UnifiedMemory_Test]+0x28): undefined reference to `testing::Test::TearDown()'
collect2: error: ld returned 1 exit status
<builtin>: recipe for target 'TestGpuIndexFlat' failed
make[1]: *** [TestGpuIndexFlat] Error 1
make[1]: Leaving directory '/home/ubuntu/faiss/gpu/test'
Makefile:94: recipe for target 'test_gpu' failed
make: *** [test_gpu] Error 2


# Platform
Ubuntu 16.04
cuda-9.0
nvidia-396
gcc-5.4.0
swig-4.0.0

Running on:
- [ ] GPU

Interface: 
- [ ] C++

# Reproduction instructions

--- a/gpu/test/Makefile
+++ b/gpu/test/Makefile
@@ -9,8 +9,10 @@ TESTS_SRC = TestGpuIndexFlat.cpp TestGpuIndexIVFPQ.cpp \
 TestGpuIndexBinaryFlat.cpp TestGpuIndexIVFFlat.cpp TestGpuMemoryException.cpp
 CUDA_TESTS_SRC = TestGpuSelect.cu
 
-TESTS_BIN = $(TESTS_SRC:.cpp=)
-CUDA_TESTS_BIN = $(CUDA_TESTS_SRC:.cu=)
+TESTS_OBJ = $(TESTS_SRC:.cpp=.o)
+CUDA_TESTS_OBJ = $(CUDA_TESTS_SRC:.cu=.o)
+
+TESTS_BIN = $(TESTS_OBJ:.o=) $(CUDA_TESTS_OBJ:.o=)
 
 
 # test_gpu_index.py test_pytorch_faiss.py
@@ -18,13 +20,13 @@ CUDA_TESTS_BIN = $(CUDA_TESTS_SRC:.cu=)
 run: $(TESTS_BIN) $(CUDA_TESTS_BIN)
 	for t in $(TESTS_BIN) $(CUDA_TESTS_BIN); do ./$$t || exit; done
 
-$(CUDA_TESTS_SRC:.cu=.o): %.o: %.cu gtest
+$(CUDA_TESTS_OBJ): %.o: %.cu gtest
 	$(NVCC) $(NVCCFLAGS) -g -O3 -o $@ -c $< -Igtest/include
 
-$(CUDA_TESTS_SRC:.cpp=.o): %.o: %.cpp gtest
+$(TESTS_OBJ): %.o: %.cpp gtest
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(CPUFLAGS) -o $@ -c $< -Igtest/include
 
-%: %.o TestUtils.o ../../libfaiss.a gtest/make/gtest.a
+$(TESTS_BIN): %: %.o TestUtils.o ../../libfaiss.a gtest/make/gtest.a
 	$(CXX) -o $@ $^ $(LDFLAGS) $(LIBS)
 
 demo_ivfpq_indexing_gpu: demo_ivfpq_indexing_gpu.o ../../libfaiss.a
@@ -46,4 +48,4 @@ clean:
 	rm -rf gtest
 	rm -f demo_ivfpq_indexing_gpu
 
-.PHONY: clean
+.PHONY: clean run

